pkg/devfile/adapters/common/utils.go

```diff
@@ -219,3 +221,37 @@ func GetSyncFilesFromAttributes(commandsMap PushCommandsMap) map[string]string {
 	}
 	return syncMap
 }
+
+// RemoveDevfileURIContents removes contents
+// which are used via a URI in the devfile
+func RemoveDevfileURIContents(devfile devfileParser.DevfileObj, componentContext string) error {
+	return removeDevfileURIContents(devfile, componentContext, devfilefs.DefaultFs{})
+}
+
+func removeDevfileURIContents(devfile devfileParser.DevfileObj, componentContext string, fs devfilefs.Filesystem) error {
+	components, err := devfile.Data.GetComponents(parsercommon.DevfileOptions{})
+	if err != nil {
+		return err
+	}
+	for _, component := range components {
+		var uri string
+		if component.Kubernetes != nil && component.Kubernetes.Uri != "" {
+			uri = component.Kubernetes.Uri
+		}
+
+		if component.Openshift != nil && component.Openshift.Uri != "" {
+			uri = component.Openshift.Uri
+		}
+
+		if uri == "" {
+			continue
+		}
+
+		completePath := filepath.Join(componentContext, uri)
```

Comment:
You should test that the URI is a relative path, and not an http[s]:// URI

Reply:
Fixed